src/main/java/com/example/CleanFolders.java

```diff
@@ -0,0 +1,127 @@
+package com.example;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * Egy könyvtárstruktúrában dokumentumokat szerkesztenek.
+ *
+ * Minden dokumentumszerkesztéskor .bak kiterjesztésű fájlok biztonsági mentés jönnek létre
+ * (doc1.doc-hoz doc1.bak), azonban gyakori, hogy a szerkesztő bezárása után ottmaradnak.
+ *
+ * Sok esetben a dokumentumokat már törlik is, és üres könyvtárra egyébként nem lenne szükség,
+ * de a .bak fájlok miatt a könyvtárak megmaradnak.
+ *
+ * A feladat egy olyan program készítése, ami a paraméterként kapott könyvtárból kiindulva
+ * kitisztítja a dokumentum nélkül ottmaradt .bak fájlokat, és az üres könyvtárakat is törli.
+ *
+ * A struktúra egy nagyon nagy (több milliós) virtuális fájlrendszer, és egy törlés viszonylag lassú művelet,
+ * ezért fontos több szálon végezni a törléseket - a fájlrendszer elosztott,
+ * ezért jól fogja bírni a párhuzamos terhelést.
+ *
+ * Megoldásként Java nyelvű forráskódot várunk (tetszőleges Java verzió használható).
+ */
+public class CleanFolders {
+	private static final List<File> deletableBaks = new ArrayList<>();
+	private static final List<File> folders = new ArrayList<>();
+
+	/**
+	 * Detects directories under the given directory and collects all .bak files without corresponding .doc files next to them.
+	 * The method uses checkFolder() for detection and loneBakDelete() for deletion.
+	 * @param startFolder	File of the starting directory.
+	 */
+	public static void bakDeletion(File startFolder) {
+		checkFolder(startFolder);
+		loneBakDelete();
+	}
+
+	/**
+	 * Returns the collected, deletable .bak files.
+	 * Created for opening a safe backdoor for testing purposes.
+	 * @return	List<File> of .bak files.
+	 */
+	public static List<File> getDeletableBaks() {
+		return deletableBaks;
+	}
+
+	/**
+	 * Returns the collected directories.
+	 * Created for opening a safe backdoor for testing purposes.
+	 * @return	List<File> of directories.
+	 */
+	public static List<File> getFolders() {
+		return folders;
+	}
+
+	/**
+	 * Helper method for directory stream filtering and collecting.
+	 * Also, launching the next recursive method call of checkFolder().
+	 * @param file	File of processed folder.
+	 */
+	private static void folderHelper(File file) {
+		folders.add(file);
+		checkFolder(file);
+	}
+
+	/**
+	 * Helper method for file filtering.
+	 * @param file	File of processed file.
+	 * @return		boolean value marks whether the checked file can be deleted.
+	 */
+	private static boolean isDeletableBak(File file) {
+		String path = file.getPath();
+		return file.isFile() &&
+				file.getName().endsWith(".bak") &&
+				!(new File(path.substring(0, path.length() - 4) + ".doc")).exists();
+	}
+
+	/**
+	 * Processes directories and files in a directory.
+	 * Recursive method for directories.
+	 * Puts all directories and only deletable files into correspondent Lists.
```

Comment:
You don't have to write down what the method does. I can read the method if I want to know what does it do. Also such comments can quickly become obsolete when someone changes the code and doesn't update the comment above. 

The "why" could be interesting here though. Write down "why" do you do not "what".

Reply:
Sorry, I read your comments FIFO :)

I thought I should document similarly as original Java methods are documented. They contain what they do plus other infos about what happens in the background. I wouldn't write the background but would write what is the purpose.

The most confortable would be not writing any docs... Although I don't think that would be appreciated.

In case a developer changes the code but not checking the docs, that would be error from his/her part, no?

Little bit confused here how a proper docs would be created nowadays :)